sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -772,6 +772,7 @@ object SQLConf {
         " produced by a builtin function such as to_char or CAST")
       .version("4.0.0")
       .stringConf
+      .checkValue({s => CollationFactory.collationNameToId(s) >= 0}, "")
```

Comment:
```suggestion
      .checkValue(s => CollationFactory.collationNameToId(s) >= 0, "")
```

Reply:
let's provide an error message

Reply:
oh, `CollationFactory.collationNameToId(s)` already throws exception?

Reply:
Yes, it gives suggestion for the name user can use.

Reply:
I could add a message here: "Collation name is valid, but is not supported as a session level default." ?

Reply:
If in future we decide to restrict the set of session level defaults.